src/fabric/src/fabric2_fdb.erl

```diff
@@ -610,6 +612,21 @@ incr_stat(#{} = Db, Section, Key, Increment) when is_integer(Increment) ->
     erlfdb:add(Tx, BinKey, Increment).
 
 
+get_active_tasks() ->
+    ActiveTasks = lists:foldl(fun(Type, TaskAcc) ->
+        Tasks = couch_jobs_fdb:tx(couch_jobs_fdb:get_jtx(undefined),
+            fun(JTx) ->
+                JobIds = couch_jobs:get_active_jobs(JTx, Type),
+                maps:map(fun(JobId, _) ->
+                    {ok, Data} = couch_jobs:get_job_data(JTx, Type, JobId),
+                    maps:get(<<"active_tasks_info">>, Data, #{})
+                end, JobIds)
+            end),
+        maps:merge(TaskAcc, Tasks)
+    end, #{}, ?ACTIVE_TASK_JOB_TYPES),
+    maps:values(ActiveTasks).
+
+
```

Comment:
Wonder if we could make a separate module for active_tasks, since most things in `fabric2_fdb` deal with db handles and we have a few indexing or db expiration modules that have the `fabric2_<...>` pattern already. 

It could be something like `fabric2_active_tasks` and it would encapsulate both the `?ACTIVE_TASK_JOB_TYPES` list, as well as be the only place that knows about <<"active_tasks_info">>.

So it might have a get_active_tasks() function and also another one that update a job's data with an active task status, like say `update_active_task_info(#{} = JobData, #{} = ActiveTaskInfo)` and it would just do `JobData#{<<"active_tasks_info">> => ActiveTaskInfo}`. Another alternative would be to have an `?ACTIVE_TASK_INFO` define in fabric2.hrl. Just a small ergonomic update to make we don't mistype <<"active_tasks_info">> name somewhere...

Reply:
@nickva created separate modules for active tasks in fabric and couch_views. I got rid of the need to have a `<<"active_tasks_info">>`. Let me know what you think.